### ISSUE TITLE
Panic in Coordinator

### ISSUE THREAD

**Issue by bonomat** [get10101/10101#541]
Logs: 

```
761/lightning/src/ln/peer_handler.rs","log.line":1693},"target":"lightning::ln::peer_handler"}
{"timestamp":"2023-04-27T21:42:38.71069033Z","level":"DEBUG","fields":{"message":"Handling SendChannelAnnouncement event in peer_handler for node 031b8396b569c55a98a006a382ebd231b4c4a2f4a93ba4aeb2e81041ca2f3e8d48 for short channel id 865292561450532864","log.target":"lightning::ln::peer_handler","log.module_path":"lightning::ln::peer_handler","log.file":"/home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/6df5761/lightning/src/ln/peer_handler.rs","log.line":1693},"target":"lightning::ln::peer_handler"}
{"timestamp":"2023-04-27T21:42:39.099463985Z","level":"DEBUG","fields":{"message":"Marking channels with 035b3080475748f1251a45c111cd4a978628aa3e0d847e8f499004a9817fa1b07f disconnected and generating channel_updates. We believe we can make future connections to this peer.","log.target":"lightning::ln::channelmanager","log.module_path":"lightning::ln::channelmanager","log.file":"/home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/6df5761/lightning/src/ln/channelmanager.rs","log.line":6553},"target":"lightning::ln::channelmanager"}
{"timestamp":"2023-04-27T21:42:39.33583433Z","level":"INFO","fields":{"message":"Found connection to target peer. Continuing HTLCIntercepted event.","target_node_id":"031b8396b569c55a98a006a382ebd231b4c4a2f4a93ba4aeb2e81041ca2f3e8d48"},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:39.336217606Z","level":"WARN","fields":{"message":"call 'estimate_fee' failed with EOF while parsing a value at line 1 column 0, retry: 1/1","log.target":"electrum_client::client","log.module_path":"electrum_client::client","log.file":"/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/electrum-client-0.12.0/src/client.rs","log.line":212},"target":"electrum_client::client"}
{"timestamp":"2023-04-27T21:42:41.336369373Z","level":"DEBUG","fields":{"message":"new_ssl socket_addrs.domain():Some(\"blockstream.info\") validate_domain:true timeout:None","log.target":"electrum_client::raw_client","log.module_path":"electrum_client::raw_client","log.file":"/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/electrum-client-0.12.0/src/raw_client.rs","log.line":321},"target":"electrum_client::raw_client"}
{"timestamp":"2023-04-27T21:42:41.387821976Z","level":"INFO","fields":{"message":"Succesfully created new client","log.target":"electrum_client::client","log.module_path":"electrum_client::client","log.file":"/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/electrum-client-0.12.0/src/client.rs","log.line":212},"target":"electrum_client::client"}
{"timestamp":"2023-04-27T21:42:42.030846548Z","level":"INFO","fields":{"message":"Started channel creation for in-flight payment","peer":"031b8396b569c55a98a006a382ebd231b4c4a2f4a93ba4aeb2e81041ca2f3e8d48","temp_channel_id":"e65ed28075a67a260f9ab1854b0236615e27de30ebe5a65bd92189ba181593ce"},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:42.030879319Z","level":"DEBUG","fields":{"message":"Successfully handled event","event":"\"HTLCIntercepted { intercept_id: InterceptId([175, 149, 22, 88, 94, 93, 205, 4, 27, 254, 226, 216, 32, 254, 11, 77, 2, 232, 173, 133, 100, 199, 129, 206, 27, 1, 17, 123, 162, 117, 68, 149]), requested_next_hop_scid: 603016157235052551, payment_hash: PaymentHash([127, 220, 38, 34, 77, 183, 140, 126, 105, 246, 87, 83, 227, 145, 185, 3, 169, 182, 34, 244, 222, 237, 237, 70, 114, 147, 252, 64, 81, 27, 44, 55]), inbound_amount_msat: 51001000, expected_outbound_amount_msat: 50000000 }\""},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:42.030906573Z","level":"DEBUG","fields":{"message":"Handling SendOpenChannel event in peer_handler for node 031b8396b569c55a98a006a382ebd231b4c4a2f4a93ba4aeb2e81041ca2f3e8d48 for channel e65ed28075a67a260f9ab1854b0236615e27de30ebe5a65bd92189ba181593ce","log.target":"lightning::ln::peer_handler","log.module_path":"lightning::ln::peer_handler","log.file":"/home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/6df5761/lightning/src/ln/peer_handler.rs","log.line":1611},"target":"lightning::ln::peer_handler"}
{"timestamp":"2023-04-27T21:42:42.14486456Z","level":"INFO","fields":{"message":"Received event","event":"HTLCIntercepted { intercept_id: InterceptId([168, 121, 235, 81, 32, 8, 18, 142, 62, 38, 18, 161, 9, 235, 138, 67, 182, 64, 93, 236, 10, 61, 30, 143, 121, 240, 229, 18, 250, 117, 135, 31]), requested_next_hop_scid: 603016157235052551, payment_hash: PaymentHash([68, 25, 139, 87, 115, 241, 5, 49, 14, 80, 71, 162, 236, 244, 219, 79, 100, 16, 149, 192, 111, 159, 199, 184, 246, 152, 233, 198, 71, 131, 246, 144]), inbound_amount_msat: 51001000, expected_outbound_amount_msat: 50000000 }"},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:42.144915492Z","level":"INFO","fields":{"message":"Intercepted HTLC","intercepted_id":"a879eb512008128e3e2612a109eb8a43b6405dec0a3d1e8f79f0e512fa75871f","requested_next_hop_scid":603016157235052551,"payment_hash":"44198b5773f105310e5047a2ecf4db4f641095c06f9fc7b8f698e9c64783f690","inbound_amount_msat":51001000,"expected_outbound_amount_msat":50000000},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:42.144928032Z","level":"INFO","fields":{"message":"Found connection to target peer. Continuing HTLCIntercepted event.","target_node_id":"031b8396b569c55a98a006a382ebd231b4c4a2f4a93ba4aeb2e81041ca2f3e8d48"},"target":"ln_dlc_node::ln::event_handler"}
{"timestamp":"2023-04-27T21:42:42.153829737Z","level":"ERROR","fields":{"message":"Aborting after panic in task","info":"panicked at 'called `Option::unwrap()` on a `None` value', /home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/6df5761/lightning/src/ln/channelmanager.rs:1816:117"},"target":"coordinator"}
   0: coordinator::main::{{closure}}::{{closure}}
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:2002:9
   2: std::panicking::rust_panic_with_hook
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:692:13
   3: std::panicking::begin_panic_handler::{{closure}}
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:577:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys_common/backtrace.rs:137:18
   5: rust_begin_unwind
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:575:5
   6: core::panicking::panic_fmt
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:64:14
   7: core::panicking::panic
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:114:5
   8: lightning::ln::channelmanager::ChannelManager<M,T,K,F,L>::list_channels
   9: ln_dlc_node::ln::event_handler::EventHandler<P>::match_event::{{closure}}
  10: tokio::runtime::park::CachedParkThread::block_on
  11: tokio::runtime::handle::Handle::block_on
  12: <ln_dlc_node::ln::event_handler::EventHandler<P> as lightning::util::events::EventHandler>::handle_event
  13: <lightning_invoice::payment::InvoicePayerUsingTime<P,R,L,E,T> as lightning::util::events::EventHandler>::handle_event
  14: <lightning::ln::channelmanager::ChannelManager<M,T,K,F,L> as lightning::util::events::EventsProvider>::process_pending_events
  15: std::sys_common::backtrace::__rust_begin_short_backtrace
  16: core::ops::function::FnOnce::call_once{{vtable.shim}}
  17: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:1988:9
  18: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/alloc/src/boxed.rs:1988:9
  19: std::sys::unix::thread::Thread::new::thread_start
             at ./rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/sys/unix/thread.rs:108:17
  20: <unknown>
  21: __clone
```

Happened on mainnet

**Comment by holzeis** [comment]
The panic is thrown in the rust-lightning library. There are a couple of unwraps, which I guess are reasonable.

```
let channel_state = self.channel_state.lock().unwrap();
let best_block_height = self.best_block.read().unwrap().height();
```

After shortly discussing it with @klochowicz, we think the best course of action is to be resilient to these panics and restart the coordinator when this happens.

We might want to open a bug in rust-lightning, if this is still happening after we have upgraded to the latest version.

On main our coordinator did not restart, because we did not start it with systemctl, but with a regular `docker run command`.
